packages-exp/auth-exp/src/core/user/user_impl.ts

```diff
@@ -97,8 +98,15 @@ export class UserImpl implements User {
     return reload(this);
   }
 
-  delete(): Promise<void> {
-    throw new Error('Method not implemented.');
+  async delete(): Promise<void> {
+    const idToken = await this.getIdToken();
+    await deleteAccount(this.auth, { idToken });
+    this.stsTokenManager.refreshToken = null;
```

Comment:
can you make a method on stsTokenManager that resets it? you're reaching into internal implementation details here

Reply:
Done